### PR TITLE
(PDB-2366) PQL tutorial documentation

### PR DESCRIPTION
**Pull request by kbarber**
This commit defines a new PQL specific tutorial.

Signed-off-by: Ken Barber <ken@bob.sh>

### CODE REVIEW

File: documentation/api/query/tutorial-pql.markdown [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+
+Now suppose we want to find the user "nick" on every node. We can use this query:
+
+    resources { type = "User" and title = "nick" }
+
+This query has two `"="` clauses, both of which must be true.
+
+The `"="` operator follows a specific structure:
+
+`<attribute to compare> = <value>`
```
Comment (ajroetker) [inline documentation/api/query/tutorial-pql.markdown]: Should this be indented as well?

File: documentation/api/query/tutorial-pql.markdown [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+After all, if we're repeating ourselves, something may be wrong! Fortunately,
+there's an operator to help us:
+
+    resources {
+      type = "User" and
+      title = "nick" and
+      !(file = "/etc/puppetlabs/code/environments/production/manifests/user.pp" and line = 111)
+    }
+
+The `!` operator wraps another clause, and returns results for which the
```
Comment (ajroetker) [inline documentation/api/query/tutorial-pql.markdown]: Up above you quote and backtick the operators. Minor nitpick but could we align them so they're all in the same format?

File: documentation/api/query/tutorial-pql.markdown [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+Let's start with `resources[certname] { <filter> }`. This query would return the
+certname for the results that match the filter as specified.
+
+We then use an operator called "in" to turn our list of resources into
+just a list of certnames. So we now conceptually have something like:
+
+    certname in ["foo.example.com", "bar.example.com", "baz.example.com"]
+
+The "in" operator matches facts whose "certname" is in the supplied list. (In our case, this list is generated by a subquery. To use a literal list, you must use the the "list" syntax described in the [PQL lists][lists].)
```
Comment (ajroetker) [inline documentation/api/query/tutorial-pql.markdown]: I think this "in" needs some backtick loving `\`"in"\``

File: documentation/api/query/tutorial-pql.markdown [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+certname? We can, with this fact query:
+
+    facts {
+      name = "ipaddress" and
+      certname in resources[certname] { type = "Class" and title = "apache" }
+    }
+
+This may appear a little daunting, so we'll look at it piece by piece.
+
+Let's start with `resources[certname] { <filter> }`. This query would return the
```
Comment (ajroetker) [inline documentation/api/query/tutorial-pql.markdown]: s/would/will

File: documentation/api/query/tutorial-pql.markdown [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+certname for the results that match the filter as specified.
+
+We then use an operator called "in" to turn our list of resources into
+just a list of certnames. So we now conceptually have something like:
+
+    certname in ["foo.example.com", "bar.example.com", "baz.example.com"]
+
+The "in" operator matches facts whose "certname" is in the supplied list. (In our case, this list is generated by a subquery. To use a literal list, you must use the the "list" syntax described in the [PQL lists][lists].)
+
+### Implicit subqueries
```
Comment (ajroetker) [inline documentation/api/query/tutorial-pql.markdown]: Might be nice to have links to the more thorough documentation for each section in here :-)

File: documentation/api/query/tutorial-pql.markdown [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+a particular set of keys.
+
+* `parameters`: this field is itself an object, containing all the parameters and values of the resource
+* `line`: the line the resource was declared on
+* `file`: the file the resource was specified in
+* `exported`: true if the resource was exported by this node, or false otherwise
+* `tags`: all the tags on the resource
+* `title`: the resource title
+* `type`: the resource type
+* `resources`: this is an internal identifier for the resource used by PuppetDB
```
Comment (wkalt) [inline documentation/api/query/tutorial-pql.markdown]: should be resource

File: documentation/api/query/tutorial-pql.markdown [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+a "value". As you might expect, we can query using "name" or "value".
+
+    facts {
+      name = "operatingsystem" and
+      value = "Debian"
+    }
+
+This will find all the "operatingsystem = Debian" facts, and their
+corresponding nodes. As you see, "and" is supported for facts, as are "or" and
+"not".
```
Comment (wkalt) [inline documentation/api/query/tutorial-pql.markdown]: pretty minor but we've got quotes on operators here and backticks on the next line

**Comment by wkalt** [comment]
+1 aside from outstanding comments

File: documentation/api/query/tutorial-pql.markdown (hunk 2 above)
Comment (kbarber) [inline documentation/api/query/tutorial-pql.markdown]: I've got backtick crazy now, just for you :-)

File: documentation/api/query/tutorial-pql.markdown (hunk 5 above)
Comment (kbarber) [inline documentation/api/query/tutorial-pql.markdown]: I've done that across the document now.

**Pull request merged**